### PR TITLE
Add ACR publish tests

### LINKED ISSUE #1559: Add more tests for ACR
### Summary of the new feature / enhancement

- Install tests for ACR
- Publish tests for ACR

### Proposed technical implementation details (optional)

_No response_

### PR DESCRIPTION
**Pull request by alerickson**
# PR Summary
Adding test suite for Publish-PSResource from ACR server.

Also resolves #1559 

## PR Checklist

- [ ] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [ ] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [ ] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

**Review by anamnavi — APPROVED** [review]
LGTM

File: test/PublishPSResourceTests/PublishPSResourceACRServer.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,452 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+$modPath = "$psscriptroot/../PSGetTestUtils.psm1"
+Import-Module $modPath -Force -Verbose
+
```
Comment (adityapatwardhan) [inline test/PublishPSResourceTests/PublishPSResourceACRServer.Tests.ps1:6]: extra line
```suggestion
```

File: test/PublishPSResourceTests/PublishPSResourceACRServer.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+        $script:DependencyModuleName = "TEMP-PackageManagement"
+        $script:DependencyModuleBase = Join-Path $script:tmpModulesPath -ChildPath $script:DependencyModuleName
+        if(!(Test-Path $script:DependencyModuleBase))
+        {
+            New-Item -Path $script:DependencyModuleBase -ItemType Directory -Force
+        }
+
+        # Create temp destination path
+        $script:destinationPath = [IO.Path]::GetFullPath((Join-Path -Path $TestDrive -ChildPath "tmpDestinationPath"))
+        New-Item $script:destinationPath -ItemType directory -Force
```
Comment (adityapatwardhan) [inline test/PublishPSResourceTests/PublishPSResourceACRServer.Tests.ps1]:
```suggestion
        $null = New-Item $script:destinationPath -ItemType directory -Force
```

File: test/PublishPSResourceTests/PublishPSResourceACRServer.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+
+        #Create folder where we shall place all script files to be published for these tests
+        $script:tmpScriptsFolderPath = Join-Path -Path $TestDrive -ChildPath "tmpScriptsPath"
+        if(!(Test-Path $script:tmpScriptsFolderPath))
+        {
+            New-Item -Path $script:tmpScriptsFolderPath -ItemType Directory -Force
```
Comment (adityapatwardhan) [inline test/PublishPSResourceTests/PublishPSResourceACRServer.Tests.ps1]:
```suggestion
            $null = New-Item -Path $script:tmpScriptsFolderPath -ItemType Directory -Force
```

**Review by adityapatwardhan — APPROVED** [review]

**Comment by alerickson** [comment]
closing this PR for #1571

**Pull request closed without merging**